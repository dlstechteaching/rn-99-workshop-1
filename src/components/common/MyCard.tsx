import React from 'react';
import { Ionicons } from '@expo/vector-icons';
import { TouchableOpacity, StyleSheet, View, Text } from 'react-native';

interface MyCardProps {
	type: string;
	title: string;
	desc: string;
	badges: string[];
	comments: string[];
}

export const MyCard: React.VFC<MyCardProps> = ({ type, title, desc, badges, comments }) => {
	return (
		<View style={styles.container}>
			<View style={styles.header}>
				<Text style={styles.type}>{type.toUpperCase()}</Text>
				<Text style={styles.title} numberOfLines={2}>
					{title}
				</Text>
			</View>
			<View style={styles.badges}>
				{badges.map((badge, i) => (
					<View style={styles.badge} key={i}>
						<Text style={styles.badgeText}>{badge.toUpperCase()}</Text>
					</View>
				))}
			</View>
			<Text style={styles.desc}>{desc}</Text>

			<View style={styles.footer}>
				<Text style={styles.comments}>{comments.length + ' comments'} </Text>
				<TouchableOpacity style={styles.button}>
					<Ionicons name={'md-arrow-forward'} size={18} color="green" />
				</TouchableOpacity>
			</View>
		</View>
	);
};

const styles = StyleSheet.create({
	container: {
		backgroundColor: 'white',
		padding: 20,
		borderRadius: 20,
		width: 325,
		flexDirection: 'column',
		marginRight: 20,
		justifyContent: 'space-between',
	},
	header: {
		width: '70%',
		marginBottom: -40,
	},
	type: {
		color: 'rgb(0, 148, 139)',
	},
	title: {
		marginBottom: 20,
	},
	badges: {
		flexDirection: 'row',
	},
	badge: {
		paddingVertical: 5,
		paddingHorizontal: 10,
		backgroundColor: 'rgb(255, 242, 215)',
		marginRight: 5,
		borderRadius: 10,
	},
	badgeText: {
		color: 'rgb(255, 175, 77)',
	},
	desc: {
		marginTop: -30,
		width: '80%',
		color: 'rgb(152, 152, 152)',
	},
	footer: {
		flexDirection: 'row',
		alignItems: 'center',
		justifyContent: 'space-between',
	},
	comments: {
		color: 'rgb(152, 152, 152)',
	},
	button: {
		borderRadius: 25,
		flexShrink: 1,
		backgroundColor: 'rgb(236, 248, 248)',
		height: 45,
		width: 45,
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
	},
});
