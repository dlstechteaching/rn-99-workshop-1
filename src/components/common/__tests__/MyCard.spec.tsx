import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { MyCard } from '../MyCard';

describe('MyCard tests', () => {
	let component: ShallowWrapper;

	it('should render properly', () => {
		givenComponent();
		thenItRendersProperly();
	});

	function givenComponent() {
		component = shallow(
			<MyCard 
				type={'label'} 
				title={'title'} 
				desc={'desc'} 
				badges={['test']} 
				comments={['test', 'test']} />
		);
	}

	function thenItRendersProperly() {
		expect(component).toMatchSnapshot();
	}

});