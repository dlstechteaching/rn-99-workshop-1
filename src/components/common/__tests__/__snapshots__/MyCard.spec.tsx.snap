// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCard tests should render properly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 20,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "marginRight": 20,
      "padding": 20,
      "width": 325,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": -40,
        "width": "70%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(0, 148, 139)",
        }
      }
    >
      LABEL
    </Text>
    <Text
      numberOfLines={2}
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      title
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      key="0"
      style={
        Object {
          "backgroundColor": "rgb(255, 242, 215)",
          "borderRadius": 10,
          "marginRight": 5,
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(255, 175, 77)",
          }
        }
      >
        TEST
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "rgb(152, 152, 152)",
        "marginTop": -30,
        "width": "80%",
      }
    }
  >
    desc
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(152, 152, 152)",
        }
      }
    >
      2 comments
       
    </Text>
    <ForwardRef
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(236, 248, 248)",
          "borderRadius": 25,
          "display": "flex",
          "flexShrink": 1,
          "height": 45,
          "justifyContent": "center",
          "width": 45,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="green"
        name="md-arrow-forward"
        size={18}
      />
    </ForwardRef>
  </View>
</View>
`;
